app/models/cherrypick_task/control_locator.rb

```diff
@@ -29,23 +29,35 @@ class CherrypickTask::ControlLocator
   # limit ourself to primes to simplify validation.
   BETWEEN_PLATE_OFFSETS = [53, 59].freeze
 
-  attr_reader :batch_id, :total_wells, :wells_to_leave_free, :num_control_wells, :available_positions
+  attr_reader :batch_id,
+              :total_wells,
+              :wells_to_leave_free,
+              :num_control_wells,
+              :available_positions,
+              :control_source_plate
 
   # @note wells_to_leave_free was originally hardcoded for 96 well plates at 24, in order to avoid
   # control wells being missed in cDNA quant QC. This requirement was removed in
   # https://github.com/sanger/sequencescape/issues/2967 however I've avoided stripping out the behaviour
   # completely in case controls are used in other pipelines.
   #
-  # @param batch_id [Integer] The id of the batch, used to generate a starting position
-  # @param total_wells [Integer] The total number of wells on the plate
-  # @param num_control_wells [Integer] The number of control wells to lay out
-  # @param wells_to_leave_free [Enumerable] Array or range indicating the wells to leave free from controls
-  def initialize(batch_id:, total_wells:, num_control_wells:, wells_to_leave_free: [])
-    @batch_id = batch_id
-    @total_wells = total_wells
-    @num_control_wells = num_control_wells
-    @wells_to_leave_free = wells_to_leave_free.to_a
-    @available_positions = (0...total_wells).to_a - @wells_to_leave_free
+  # @param params [Hash] A hash containing the following keys:
+  #   - :batch_id [Integer] The id of the batch, used to generate a starting position
+  #   - :total_wells [Integer] The total number of wells on the plate
+  #   - :num_control_wells [Integer] The number of control wells to lay out
+  #   - :wells_to_leave_free [Enumerable] Array or range indicating the wells to leave free from controls
+  #   - :available_positions [Enumerable] Array or range indicating the available positions for controls
```

Comment:
I don't think this is something you want in the params hash.  It's calculated by removing the `:wells_to_leave` from the full set of wells.